portfolio/src/main/webapp/script.js

```diff
@@ -39,4 +43,17 @@ function deleteComments() {
     console.log("fetching deletion");
     fetch("/delete-data", { method: "post" }).then(response => console.log(response.text()));
     getComments();
+}
+
+function printCommentSender() {
+    fetch("/Authenticate").then(userData => userData.json()).then(nicknameJson => {
+        console.log(nicknameJson);
+        commentSenderDiv = document.getElementById("comment-sender-div");
+        commentSenderDiv.innerHTML =
+        `<p>Posting as <b>${nicknameJson.nickname}</b>. visit <a class="hyperlink" href="/settings">Settings</a> to update nickname.</p>
```

Comment:
For adding content to a page, it's good practice to avoid setting innerHTML to anything even remotely user-entered to avoid any cross site scripting issues (https://en.wikipedia.org/wiki/Cross-site_scripting). While it's probably OK for a small side project, you may want to try constructing the DOM elements and using `appendChild `instead (https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild).